.NET CLI Build Binary log filenames aren't quoted correctly

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

1.2.0

### Operating system

Linux, Windows, macOS

### Operating system architecture

64-Bit

### CI Server

Azure Pipelines

### What are you seeing?

Build is failing @Eilon 

```
Setting environment variable: PATH = '/Users/builder/azdo/_work/1/s/bin/dotnet:/Users/builder/azdo/_work/1/s/bin/dotnet:/usr/local/microsoft/powershell/7:~/Library/Developer/Xamarin/android-sdk-macosx:/Library/Frameworks/Mono.framework/Versions/Current/Commands/:/usr/local/bin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/bin:/bin:/usr/sbin:/sbin'
Executing: /Users/builder/azdo/_work/1/s/bin/dotnet/dotnet build "./templatesTest/maui Space-Dash" /property:configuration=Release /property:RestoreNoCache=true /property:RestorePackagesPath=/Users/builder/azdo/_work/1/s/templatesTest/packages /property:RestoreConfigFile=/Users/builder/azdo/_work/1/s/templatesTest/nuget.config /binarylogger:/Users/builder/azdo/_work/1/a/logs/maui Space-Dash-Release.binlog
Microsoft (R) Build Engine version 17.0.0-preview-21418-01+46d8f9b0b for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1008: Only one project can be specified.
Switch: Space-Dash-Release.binlog

```

_Originally posted by @rmarinho in https://github.com/dotnet/maui/issues/2410#issuecomment-914525486_



> Yeah it looks like the bug fix that was in Cake doesn't fix this particular scenario 😢 . I'll close this PR and see if I can get a fix > into Cake. There seem to be tons of inconsistencies in how Cake handles quotes.
>
 _Originally posted by @Eilon in https://github.com/dotnet/maui/issues/2410#issuecomment-914546147_



### What is expected?

Bin log to be quoted

### Steps to Reproduce

```diff
-    var binlog = $"\"{logDirectory}/{name}-{configuration}.binlog\"";
+    var binlog = $"{logDirectory}/{name}-{configuration}.binlog";
   // If we're not on Windows, use ./bin/dotnet/dotnet

    if (!IsRunningOnWindows() || deployAndRun)
    {
        var msbuildSettings = new DotNetCoreMSBuildSettings()
            .SetConfiguration(configuration)
            .EnableBinaryLogger(binlog);
        if (properties != null)
        {
            foreach (var property in properties)
            {
                msbuildSettings.WithProperty(property.Key, property.Value);
            }
        }
        if (deployAndRun)
            msbuildSettings.WithTarget("Run");
        var dotnetBuildSettings = new DotNetCoreBuildSettings
        {
            MSBuildSettings = msbuildSettings,
        };
        if (localDotnet)
            dotnetBuildSettings.ToolPath = dotnetPath;
        DotNetCoreBuild(sln, dotnetBuildSettings);
```

### Output log

MSBUILD : error MSB1008: Only one project can be specified.

--- a/src/Cake.Common.Tests/Unit/Tools/DotNetCore/MSBuild/DotNetCoreMSBuildBuilderTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/DotNetCore/MSBuild/DotNetCoreMSBuildBuilderTests.cs
@@ -1295,8 +1295,8 @@ namespace Cake.Common.Tests.Unit.Tools.DotNetCore.MSBuild
             [InlineData(false, null, MSBuildBinaryLoggerImports.Unspecified, "msbuild")]
             [InlineData(true, null, MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger")]
             [InlineData(true, null, MSBuildBinaryLoggerImports.None, "msbuild /binarylogger:ProjectImports=None")]
-            [InlineData(true, "mylog.binlog", MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger:mylog.binlog")]
-            [InlineData(true, "mylog.binlog", MSBuildBinaryLoggerImports.Embed, "msbuild /binarylogger:mylog.binlog;ProjectImports=Embed")]
+            [InlineData(true, "mylog.binlog", MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger:\"mylog.binlog\"")]
+            [InlineData(true, "mylog.binlog", MSBuildBinaryLoggerImports.Embed, "msbuild /binarylogger:\"mylog.binlog\";ProjectImports=Embed")]
             public void Should_Append_Binary_Logging_If_Specified(bool enabled, string fileName, MSBuildBinaryLoggerImports imports, string args)
             {
                 // Given
@@ -1318,8 +1318,8 @@ namespace Cake.Common.Tests.Unit.Tools.DotNetCore.MSBuild
             [Theory]
             [InlineData(null, MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger")]
             [InlineData(null, MSBuildBinaryLoggerImports.None, "msbuild /binarylogger:ProjectImports=None")]
-            [InlineData("mylog.binlog", MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger:mylog.binlog")]
-            [InlineData("mylog.binlog", MSBuildBinaryLoggerImports.Embed, "msbuild /binarylogger:mylog.binlog;ProjectImports=Embed")]
+            [InlineData("mylog.binlog", MSBuildBinaryLoggerImports.Unspecified, "msbuild /binarylogger:\"mylog.binlog\"")]
+            [InlineData("mylog.binlog", MSBuildBinaryLoggerImports.Embed, "msbuild /binarylogger:\"mylog.binlog\";ProjectImports=Embed")]
             public void Should_Append_Binary_Logging_If_Enabled(string fileName, MSBuildBinaryLoggerImports imports, string args)
             {
                 // Given

--- a/src/Cake.Common/Tools/DotNetCore/MSBuild/MSBuildArgumentBuilderExtensions.cs
+++ b/src/Cake.Common/Tools/DotNetCore/MSBuild/MSBuildArgumentBuilderExtensions.cs
@@ -110,7 +110,7 @@ namespace Cake.Common.Tools.DotNetCore.MSBuild
                 string binaryOptions = null;
                 if (!string.IsNullOrEmpty(settings.BinaryLogger.FileName))
                 {
-                    binaryOptions = settings.BinaryLogger.FileName;
+                    binaryOptions = settings.BinaryLogger.FileName.Quote();
                 }
 
                 if (settings.BinaryLogger.Imports != MSBuildBinaryLoggerImports.Unspecified)

